server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -641,8 +641,10 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
     static ClusterState innerRemoveIndexTemplateV2(ClusterState currentState, String name) {
```

Comment:
Looks like name should be a string[], and also the name field on request class. Otherwise  the request can't be introspected correctly by security.